Filters/Average.cs

```diff
@@ -53,31 +66,56 @@ public Average(int FilterCount = 10)
         /// <param name="Input"> Value to feed into the filter. </param>
         public void Feed(T Input)
         {
-            // Increase number of iterations by 1
-            this.Iterations++;
+            // Value to divide sum by at the end of the computation
+            int Divisor = this.Iterations == 0 ? 1 : this.Iterations;
+
             // Store input as a dynamic type since we know T is a numeric
             dynamic dynamicInput = Input;
-            // Subtract current array index value from sum
-            this.CurSum -= this.AverageArray[this.Index];
+
             // Add current value to sum
             this.CurSum += dynamicInput;
-            // Store curent value in old spot
-            this.AverageArray[this.Index] = dynamicInput;
-            // Increment index. Go back to zero if index + 1 == filterCount
-            this.Index = (this.Index + 1) % this.FilterCount; // Increment index. Go back to zero if index + 1 == filterCount
+
+            // Check if in continuous or roll mode
+            if (this.FilterCount != null)
+            {
+                // Subtract current array index value from sum
```

Comment:
Personally, I prefer to have the comment at the end of the line, instead of taking up 3 lines per statement, like this:
`this.CurSum -= this.AverageArray[this.Index]; // Subtract current array index value from sum`

Reply:
Some of the comments would be too long I think, and I'd like them all to be organized the same, so I think I am going to leave them.